### DISCUSSION TITLE
Is there something wrong in the "colour.plotting.plot_RGB_chromaticities_in_chromaticity_diagram" definition?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#734] [Q&A] [answered]
Hi Colour Team,

I found something strange inside def plot_RGB_chromaticities_in_chromaticity_diagram:
https://github.com/colour-science/colour/blob/c9a758d7f807217dfca799f33aa9d137f04ca445/colour/plotting/models.py#L738

Why you guys use the same illuminant (colourspace.whitepoint) for both source and destination colour space?

Best,
Hoang

**Comment by thomasmansencal** [comment, marked as answer]
Hi @hminle,

It might seem odd indeed but this is correct: Given a RGB array encoded in a given colourspace, e.g. ProPhoto RGB, we want to see its chromaticity coordinates as encoded by that colourspace. The conversion from RGB to XYZ needs to be done in a way where no whitepoint change happens, this is to say that the whitepoint for plotting the chromaticities should be that of the RGB array, i.e. D50 for ProPhoto RGB. By passing the given colourspace whitepoint two times to the `colour.RGB_to_XYZ` definition, we are ensuring that this is the case and that no chromatic adaptation occurs.

Note that the displayed colours, however, ARE converted to the plotting colourspace, i.e. sRGB by default and thus there will be a whitepoint change in that case.

Hope that helps!

Cheers,

Thomas

**Comment by hminle** [comment]
Oh thank you, now I got it
